packages/api/app/index.js

```diff
@@ -13,10 +13,35 @@ const awsServerlessExpressMiddleware = require('aws-serverless-express/middlewar
 
 const { services } = require('@cumulus/aws-client');
 const { MissingRequiredEnvVar } = require('@cumulus/errors');
+const { secretsManager } = require('@cumulus/aws-client/services');
+const Logger = require('@cumulus/logger');
 
 const router = require('./routes');
 const { jsonBodyParser } = require('./middleware');
 
+const log = new Logger({ sender: '@api/index' });
+
+// Load Environment Variables
+// This should be done outside of the handler to minimize Secrets Manager calls.
+const initEnvVarsFunction = async () => {
+  if (!process.env.api_config_secret_id) {
+    log.warn('No process.env.api_config_secret_id set, *not* attempting to provide environment variable defaults');
+    return;
```

Comment:
Would throwing an error be inappropriate here? Wouldn't we expect most if not all API functionality to fail if necessary environment variables are missing? If so, wouldn't immediately failing the API request at this point be preferable?

Reply:
Hmm, fair point.  I was intending to allow more flexibility in our terraform configuration without requiring a lambda code change/re-deployment, but in practice, it probably doesn't make sense to not throw here.         I'll update it.